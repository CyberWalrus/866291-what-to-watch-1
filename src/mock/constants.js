const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
const GENRE_DEFOULT = `All genres`;
const NUMBER_FILM = 20;

export {FilmRoute, GENRE_DEFOULT, NUMBER_FILM};

