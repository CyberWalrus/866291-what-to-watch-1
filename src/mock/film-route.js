const FilmRoute = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export default FilmRoute;
