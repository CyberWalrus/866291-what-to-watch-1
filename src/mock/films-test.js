const films = [
  {
    id: 1,
    genre: `drama`,
    title: `Aviator`,
    src: `img/aviator.jpg`
  },
  {
    id: 2,
    genre: `drama`,
    title: `Macbeth`,
    src: `img/macbeth.jpg`
  }
];
export default films;
