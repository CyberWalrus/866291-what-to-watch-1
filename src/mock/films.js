const films = [
  {
    id: 1,
    genre: `drama`,
    title: `Aviator`,
    src: `img/aviator.jpg`
  },
  {
    id: 2,
    genre: `drama`,
    title: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    id: 3,
    genre: `drama`,
    title: `Revenant`,
    src: `img/revenant.jpg`
  },
  {
    id: 4,
    genre: `comedy`,
    title: `Johnny English`,
    src: `img/johnny-english.jpg`
  },
  {
    id: 5,
    genre: `comedy`,
    title: `What We Do in the Shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`
  }
];
export default films;
