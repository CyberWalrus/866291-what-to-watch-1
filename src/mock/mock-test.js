const FILMS = [
  {
    bgColor: `#977461`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
    director: `Martin Scorsese`,
    genre: `Thriller`,
    id: 1,
    isFavorite: false,
    pageUrl: `Shutter Island`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: `8,1`,
    ratingLevel: `Very good`,
    released: 2010,
    runTime: 138,
    scoresCount: 1002557,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
    starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
    title: `Shutter Island`
  },
  {
    bgColor: `#F1E9CE`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    director: `Justin Kurzel`,
    genre: `Drama`,
    id: 2,
    isFavorite: false,
    pageUrl: `Macbeth`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: `6,6`,
    ratingLevel: `Good`,
    released: 2015,
    runTime: 113,
    scoresCount: 48798,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Macbeth.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg`,
    starrings: [(`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`)],
    title: `Macbeth`
  }
];
const FILM = {
  bgColor: `#977461`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
  director: `Martin Scorsese`,
  genre: `Thriller`,
  id: 1,
  isFavorite: false,
  pageUrl: `Shutter Island`,
  srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: `8,1`,
  ratingLevel: `Very good`,
  released: 2010,
  runTime: 138,
  scoresCount: 1002557,
  srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
  srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
  srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
  starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
  title: `Shutter Island`
};
const REVIEWS = [
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  },
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  }
];

const GENRES = [
  `All genres`,
  `Thriller`,
  `Drama`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Action`,
  `Fantasy`
];
export {FILMS, FILM, GENRES, REVIEWS};
