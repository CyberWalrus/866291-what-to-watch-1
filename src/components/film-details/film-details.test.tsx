import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import FilmDetails from "./film-details";
import {FILM} from "../../mock/data-mock";

Enzyme.configure({adapter: new Adapter()});
it(`FilmDetails correctly renders after relaunch`, (): void => {
  const tree = toJson(
    shallow(
      <FilmDetails
        released={FILM.released}
        runTime={FILM.runTime}
        genre={FILM.genre}
        director={FILM.director}
        starrings={FILM.starrings}
      />
    )
  );
  expect(tree).toMatchSnapshot();
});
