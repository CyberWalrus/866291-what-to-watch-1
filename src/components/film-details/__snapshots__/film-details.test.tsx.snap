// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails correctly renders after relaunch 1`] = `
<Fragment>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Martin Scorsese
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Leonardo DiCaprio
          <br />
          Emily Mortimer
          <br />
          Mark Ruffalo
          <br />
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          1h 44m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Thriller
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2010
        </span>
      </p>
    </div>
  </div>
</Fragment>
`;
